File: python/generators/pairsum.py
def getPairBySum(nums,targetSum):
    """ Finds two consecutive entries in a list (it doesn’t have to be unique) that sum to a specified value.

        Parameters
        ----------
        nums : list
            A 1 dimensional list of numbers
        targetSum : number
            A number representing the target sum that we are searching for in the array of nums
    """

    #[:-1] returns all the elements of the list but the last one
    for i, _ in enumerate(nums[:-1]):  #use enumerate to create an indexed pair list and iterate
                                        # through that list to get the sums of the consecutive numbers
        if nums[i] + nums[i+1]==targetSum:
            yield "({}, {})".format(nums[i], nums[i+1])
        elif list[i-1] and nums[i] + nums[i-1]==targetSum: #check to make sure the previous elment exists 
                                                    #before checking to see if the elements equal the sum
            yield "({}, {})".format(nums[i], nums[i-1])
           
    

pair = getPairBySum([1, 9, 2, 4, 7, 4], 11)
print(next(pair))
print(next(pair))
print(next(pair))
print(next(pair))

File: javascript/src/pairsum.js
/**
* Finds two consecutive entries in a list (it doesn’t have to be unique) that
* sum to a specified value.
*
* Input: A list of numbers. An example is [1, 9, 2, 4, 7, 4] and a specified
* target sum of 11.
* Output: The procedure will identify any of these pairs: (9, 2), (2, 9), 
* (4, 7), or (7, 4).
* @function getPairBySum
* @param {Array} nums A one dimensional list of numerical values. An example 
* is [1, 9, 2, 4, 7, 4]
* @param {Number} targetSum A number representing the target sum that we are 
* searching for in the array of nums
* @yield {String} A pair of numbers found in the array nums that have a sum 
* equal to targetSum.
*/

function* getPairBySum(nums, targetSum){
   for (let i = 1; i < nums.length-1; i++) {
        const prev = nums[i-1], curr = nums[i], next=nums[i+1];
        if(prev+curr === targetSum){
            yield `(${prev}, ${curr})`;
        } else if(curr+next === targetSum){
            yield `(${curr}, ${next})`;
        }
    
   }
}



module.exports = getPairBySum